podpac/core/data/datasource.py

```diff
@@ -281,6 +281,24 @@ def _get_data(self, rc, rci):
     # Methods
     # ------------------------------------------------------------------------------------------------------------------
 
+    def get_source_data(self, bounds={}):
+        """
+        Get source data, without interpolation.
+
+        Arguments
+        ---------
+        bounds : dict
+            Dictionary of bounds by dimension, optional.
+
+        Returns
+        -------
+        data : UnitsDataArray
+            Source data
+        """
+
+        coords, I = self.coordinates.select(bounds, return_index=True)
```

Comment:
We could add a test here on the size of `I` and throw an logger warning in case of requests that exceed something like 8 GB? What do you think?

Reply:
Doesn't numpy throw it's own warning if you try to make an array that is too large?

Reply:
```
>>> np.empty((10000, 1000, 1000))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
numpy.core._exceptions.MemoryError: Unable to allocate 74.5 GiB for an array with shape (10000, 1000, 1000) and data type float64
```

Reply:
That's a good point, but I don't think all the underlying sources use numpy (I'm thinking rasterio) and the error might not get thrown at the index level since the indices might be orthogonal.